src/Microsoft.Framework.Caching.Abstractions/project.json

```diff
@@ -1,6 +1,10 @@
 {
     "dependencies": {
-        "Microsoft.Framework.NotNullAttribute.Sources": { "type": "build", "version": "1.0.0-*" }
+      "Microsoft.Framework.NotNullAttribute.Sources": {
+        "type": "build",
+        "version": "1.0.0-*"
+      },
+      "Newtonsoft.Json": "6.0.6"
```

Comment:
Totally agree. I only added the Json.NET reference because the BinaryFormatter was not available on .NET Core and using the Json serializer seemed a decent alternative.

Also, even if we do need to add Json.NET, what would be a good place to add these extension methods? There isn't a common caching implementation project, all the projects are specific implementations e.g. memory, redis, sqlserver.